showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITIam.java

```diff
@@ -0,0 +1,256 @@
+ * limitations under the License.
+ */
+package com.google.showcase.v1beta1.it;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertThrows;
+
+import com.google.api.gax.rpc.InvalidArgumentException;
+import com.google.api.gax.rpc.NotFoundException;
+import com.google.common.collect.ImmutableList;
+import com.google.iam.v1.Binding;
+import com.google.iam.v1.GetIamPolicyRequest;
+import com.google.iam.v1.Policy;
+import com.google.iam.v1.SetIamPolicyRequest;
+import com.google.iam.v1.TestIamPermissionsRequest;
+import com.google.iam.v1.TestIamPermissionsResponse;
+import com.google.showcase.v1beta1.IdentityClient;
+import com.google.showcase.v1beta1.it.util.TestClientInitializer;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ITIam {
+  private static final Policy DEFAULT_POLICY =
+      Policy.newBuilder()
+          .addBindings(Binding.newBuilder().setRole("foo.editor").addMembers("allUsers"))
+          .build();
+  private static IdentityClient grpcClient;
+  private static IdentityClient httpjsonClient;
+  private String resourceName;
+
+  @BeforeClass
+  public static void createClients() throws Exception {
+    grpcClient = TestClientInitializer.createGrpcIdentityClient();
+    httpjsonClient = TestClientInitializer.createHttpJsonIdentityClient();
+  }
+
+  @Before
+  public void setupTests() {
+    resourceName = "users/" + UUID.randomUUID().toString().substring(0, 8);
+  }
+
+  @AfterClass
+  public static void destroyClients() throws InterruptedException {
+    grpcClient.close();
+    httpjsonClient.close();
+
+    grpcClient.awaitTermination(TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+    httpjsonClient.awaitTermination(
+        TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+  }
+
+  @Test
+  public void testGrpc_setIamPolicy() {
+    SetIamPolicyRequest policyRequest =
+        SetIamPolicyRequest.newBuilder()
+            .setPolicy(DEFAULT_POLICY)
+            .setResource(resourceName)
+            .build();
+    Policy policy = grpcClient.setIamPolicy(policyRequest);
+    assertThat(policy).isEqualTo(DEFAULT_POLICY);
+  }
+
+  @Test
+  public void testHttpJson_setIamPolicy() {
+    SetIamPolicyRequest policyRequest =
+        SetIamPolicyRequest.newBuilder()
+            .setPolicy(DEFAULT_POLICY)
+            .setResource(resourceName)
+            .build();
+    Policy policy = httpjsonClient.setIamPolicy(policyRequest);
+    assertThat(policy).isEqualTo(DEFAULT_POLICY);
+  }
+
+  @Test
+  public void testGrpc_setIamPolicy_missingResource() {
+    SetIamPolicyRequest setIamPolicyRequest = SetIamPolicyRequest.newBuilder().build();
+    assertThrows(
+        InvalidArgumentException.class, () -> grpcClient.setIamPolicy(setIamPolicyRequest));
```

Comment:
Where is this exception being thrown? Is it in our client libraries? I see that we are adding a lot of error scenarios in this IT, a general comment is that we only want to test the exception if it's being thrown in the IAM code before it reaches backend, or if there is any specific response exception mapping logic in IAM code.

Reply:
This is being thrown in the backend: https://github.com/googleapis/gapic-showcase/blob/b146f892d98ea0d733c45a2315941e8e6ab69c90/server/services/iam_policy_service.go#L27-L29

There is only a few validation checks that the backend iam server does.